float32 duration
bool direction
---
